src/@next/components/organisms/AdyenPaymentGateway/AdyenPaymentGateway.tsx

```diff
@@ -0,0 +1,183 @@
+      script.integrity = scriptConfig.integrity;
+      script.crossOrigin = scriptConfig.crossOrigin;
+      script.async = true;
+      script.onload = initAdyenGatewayHandlers; // Wait until the script is loaded before initiating AdyenCheckout
+      document.body.appendChild(script);
+    }
+  }, [adyenClientKey, parsedAdyenConfig, gatewayRef.current]);
+
+  const initAdyenGatewayHandlers = () => {
+    const configuration = adyenClientKey &&
+      adyenConfig && {
+        locale: navigator.language,
+        environment: "test",
+        clientKey: adyenClientKey,
+        paymentMethodsResponse: parsedAdyenConfig,
+        showPayButton: false,
+        onSubmit: (state: any, dropin: any) => {
+          if (!state?.isValid) {
+            onError([new Error("Invalid payment submission")]);
+          } else {
+            submitPayment(state?.data)
+              .then(value => {
+                if (value.error) {
+                  onError([value.error]);
+                } else if (!value?.confirmationNeeded) {
+                  submitPaymentSuccess(value?.order);
+                } else {
+                  const paymentAction =
+                    value?.confirmationData &&
+                    JSON.parse(value?.confirmationData);
+                  dropin.handleAction(paymentAction);
+                }
+              })
+              .catch(error => {
+                onError([new Error(error)]);
+              });
+          }
+        },
+        onAdditionalDetails: (state: any, dropin: any) => {
+          if (!state?.isValid) {
+            onError([new Error("Invalid payment submission")]);
+          } else {
+            submitPayment(state?.data)
+              .then(value => {
+                if (value.error) {
+                  onError([value.error]);
+                } else if (!value?.confirmationNeeded) {
+                  submitPaymentSuccess(value?.order);
+                } else {
+                  const paymentAction =
+                    value?.confirmationData &&
+                    JSON.parse(value?.confirmationData);
+                  dropin.handleAction(paymentAction);
+                }
+              })
+              .catch(error => {
+                onError([new Error(error)]);
+              });
+          }
+        },
+        onError: (error: any) => {
+          onError([{ message: error.error }]);
+        },
+      };
+
+    const checkout = configuration && new window.AdyenCheckout(configuration);
+    const dropinElement = checkout?.create("dropin");
+
+    if (dropinElement && !dropin && gatewayRef.current) {
+      dropinElement?.mount(gatewayRef.current);
+      setDropin(dropinElement);
+    }
+  };
+
+  useEffect(() => {
+    (formRef?.current as any)?.addEventListener("submit", () => {
+      processPayment();
+    });
+  }, [formRef]);
+
+  useEffect(() => {
+    if (dropin) {
+      (formRef?.current as any)?.addEventListener("submitComplete", () => {
```

Comment:
Where does this event come from?

Reply:
It comes from CheckoutReviewSubpage component - in other words `submitComplete` is called when a user clicks the complete checkout button (the case when gateway may require additional actions like 3d secure confirmation, otherwise completeCheckout method from SDK is called immediately - so probably such an event will need to be used also when we will be implementing 3d secure for Stripe, Braintree or other gateways).